packages/exchange/src/adapters/api.test.ts

```diff
@@ -49,39 +53,101 @@ describe('exchangeApiMaker', () => {
             getBaseUrl: jest
               .fn()
               .mockReturnValue(jest.fn().mockResolvedValue('encryptus-url')),
+            extractParamsFromBaseUrl: jest.fn(() => ({
+              access_token: 'FAKE_TOKEN',
+            })),
           },
         }
 
         const api = exchangeApiMaker({isProduction, partner}, deps)
         const result = await api.getBaseUrl(providerId)
         expect(result).toBe('encryptus-url')
       })
+
+      it('does not resolve', async () => {
+        const isProduction = true
+        const partner = 'yoroi'
+        const providerId = 'none'
+        const deps = {
+          banxaApi: {
+            getBaseUrl: jest
+              .fn()
+              .mockReturnValue(jest.fn().mockResolvedValue('banxa-url')),
+          },
+          encryptusApi: {
+            getBaseUrl: jest
+              .fn()
+              .mockReturnValue(jest.fn().mockResolvedValue('encryptus-url')),
+            extractParamsFromBaseUrl: jest.fn(() => ({
```

Comment:
I don't think it needs to be part of the API, cuz the expected return of the manager.referralLink -> is to contain, this is implementation, for me this should be extracted as a helper, and be part of the create referral link, anytime the baseUrl contains an access_token, it adds in the referral link.

Reply:
like doesn't even matter the provider, if there is a `access_token` it will include it.